bin/drop_sample_annot.py

```diff
@@ -1,142 +1,109 @@
     df.fillna("NA", inplace=True)
     df.to_csv(out_file, index=False, sep="\t")
 
 
-if __name__ == "__main__":
+def parse_args(argv=None):
+    """Define and immediately parse command line arguments."""
     parser = argparse.ArgumentParser(
         formatter_class=argparse.MetavarTypeHelpFormatter,
         description="""Generate DROP sample annotation for patients.""",
     )
-
-    parser.add_argument(
-        "--bam",
-        type=str,
-        nargs="+",
-        help="bam files for the patient",
-        required=True,
-    )
-
-    parser.add_argument(
-        "--sample",
-        type=str,
-        nargs="+",
-        help="corresponding sample name",
-        required=True,
-    )
-
-    parser.add_argument(
-        "--strandedness",
-        type=str,
-        nargs="+",
-        help="strandedness of RNA",
-        required=True,
-    )
-
-    parser.add_argument(
-        "--single_end",
-        type=str,
-        nargs="+",
-        help="is the sample paired end?",
-        required=True,
-    )
-
-    parser.add_argument(
-        "--gtf",
-        type=str,
-        help="Transcript annotation file in gtf format",
-        required=True,
-    )
-
+    parser.add_argument("--bam", type=str, nargs="+", help="bam files for the patient", required=True)
+    parser.add_argument("--samples", type=str, nargs="+", help="corresponding sample name", required=True)
+    parser.add_argument("--strandedness", type=str, nargs="+", help="strandedness of RNA", required=True)
+    parser.add_argument("--single_end", type=str, nargs="+", help="is the sample paired end?", required=True)
+    parser.add_argument("--gtf", type=str, help="Transcript annotation file in gtf format", required=True)
     parser.add_argument(
         "--count_file", type=str, help="A tsv file of gene counts for all processed samples.", required=True
     )
+    parser.add_argument("--ref_annot", type=str, help="Path to reference annotation tsv", required=True)
+    parser.add_argument("--output", type=str, help="Path to save to", required=True)
+    parser.add_argument("--version", action="version", version=SCRIPT_VERSION)
+    return parser.parse_args(argv)
 
-    parser.add_argument(
-        "--ref_annot",
-        type=str,
-        help="Path to reference annotation tsv",
-        required=True,
-    )
-
-    parser.add_argument(
-        "--output",
-        type=str,
-        help="Path to save to",
-        required=True,
-    )
 
-    args = parser.parse_args()
-    SampleAnnotation(
+def main(argv=None):
+    """Coordinate argument parsing and program execution."""
+    args = parse_args(argv)
+    get_sample_annotation(
         args.bam,
-        args.sample,
+        args.samples,
         args.strandedness,
         args.single_end,
         args.gtf,
         args.count_file,
         "drop_annotation_given_samples.tsv",
     )
     final_annot(args.count_file, args.ref_annot, args.output)
```

Comment:
Both of this functions are writing to a CSV file, i would rename them to `write_*_to_tsv`

Reply:
Done :)